### PR TITLE
Better parser

### PR DESCRIPTION
**Pull request by yflim**
To-dos:
1. Update readme and docstring(s)
2. Update tests
3. Maybe: Extract only relevant parts from `tech.ml.dataset` and drop dependency
4. Attribute authorship of said parts in #2

### CODE REVIEW

**Comment by yflim** [comment]
Major changes actually made:
1. Moved parser specification out of `options` argument into its own in `load-csv`. 
2. Simplified parser specification format (see `README.md` for usage details).
3. Decouple vector parsers from automatic parser/type inference: enable independent specification of vector parsers (for select columns) as with parsers for other types.
4. Update tests, readme, and docstring(s).
5. Extracted and adapted relevant code from `tech.ml.dataset` and dropped it from main dependencies.
6. Acknowledge authorship of said code in `README.md`.

File: src/tablehike/parse/datetime.clj [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+
+(def datetime-datatypes
+  (sorted-set :offset-date-time :zoned-date-time :local-date-time :local-date :instant))
+
+
+;;Assuming that the string will have /,-. replaced with /space
+(def ^{:doc "Local date parser patterns used to generate the local date formatter."}
+  local-date-parser-patterns
+  ["yyyy MM dd"
```
Comment (whilo) [inline src/tablehike/parse/datetime.clj:19]: Are there also formats that have dashes? I don't know what is most common, but I have also seen those.

**Review by whilo — DISMISSED** [review]
Looks good! Thanks for putting in all the effort to make it generalize *and* perform well.

File: src/tablehike/parse/datetime.clj (hunk 1 above)
Comment (yflim) [reply src/tablehike/parse/datetime.clj:19]: The code looks for and replaces a set of common characters and replaces them with a space, before parsing with those patterns: https://github.com/replikativ/tablehike/blob/2b2088aff0937f610f7e8c866b886da9837c2a91/src/tablehike/parse/datetime.clj#L30-L32

**Review by whilo — APPROVED** [review]

**Pull request merged**